Autopilot/TelemetryManager/Telemetry_Manager.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file Telemetry_Manager.c
+ * @author Jingting Liu
+ * @date January 14th, 2021
+ */
+
+#include <stdio.h>
+#include <common/mavlink.h>
+
+#include "telemetry_state_machine.h"
+
+
+
+TM_STATE_e current_state = INIT;
+
+// TODO: Add FreeRTOS specifics, return type, waits etc
+void telemetery_manager()
+{
+    while (1)
+    {
+        switch(current_state)
+        {
+            case INIT:
+            {
+                // TODO: Insert init code
+                current_state = CONNECT_WITH_PATH_MANAGER;
+                break;
+            }
+            
+            case WAIT_FOR_GROUND_DATA:
+            {
+                while (! (serial.bytesAvailable > 0))
```

Comment:
We will never poll. It is based on interrupts and DMA. If there's no data available at the time you want it, you just have to move on. This is done so that each execution of the state machine takes the same amount of time regardless of anything else. That guarantees that we can keep things synchronous.